src/com/owncloud/android/datamodel/FileDataStorageManager.java

```diff
@@ -512,7 +512,7 @@ public boolean removeFile(OCFile file, boolean removeDBData, boolean removeLocal
     
 
     public boolean removeFolder(OCFile folder, boolean removeDBData, boolean removeLocalContent) {
-        boolean success = true;
+        boolean success = false;
```

Comment:
Similarly, this breaks the condition in line 520 for input parameters ```folder != null && folder.isFolder() ```, ```removeDBData == false ```. 

Once again, I'd go with an ```else``` block in line 523.